arches/app/etl_modules/bulk_data_deletion.py

```diff
@@ -129,19 +130,34 @@ def get_sample_data(self, nodegroup_id, resourceids):
 
         return sample_data[0:5]
 
-    def delete_resources(self, userid, loadid, graphid, resourceids):
+    def delete_resources(self, userid, loadid, graphid=None, resourceids=None, verbose=False):
         result = {"success": False}
-        user = User.objects.get(id=userid)
+        deleted_count = 0
+        user = User.objects.get(id=userid) if userid else {}
         try:
-            if resourceids and graphid:
-                resources = Resource.objects.filter(graph_id=graphid).filter(pk__in=resourceids)
+            if resourceids:
+                resources = Resource.objects.filter(pk__in=resourceids)
```

Comment:
I think this slightly degrades performance if both resources and graph are provided, so I don't see the readability tradeoff as worth it, but I don't see a current code path that exercises it, so I'm not too bothered.